### PR TITLE
Issue 20453 - Fixes the test_http and test_sheets assert calls only

### PR DESCRIPTION
**Pull request by harishkrao**
The PR contains fixes for part of the issues listed in the issue #20453. It fixes the tests for the providers - Google Sheets and Simple Http only. I will be submitting more PRs for the remaining open issues for #20453.

related: #20453 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by harishkrao** [comment]
@potiuk @mik-laj if you have time, can you please review the PR? If you are busy and can suggest other reviewers, that is great too. Thank you!

**Comment by harishkrao** [comment]
> @potiuk @mik-laj @kaxil if you have time, can you please review the PR? If you are busy and can suggest other reviewers, that is great too. Thank you!
@potiuk Just wanted to bump this up, when you have a moment.

File: tests/providers/google/suite/operators/test_sheets.py [hunk 1]
```diff
@@ -39,12 +39,9 @@ def test_execute(self, mock_xcom, mock_hook):
...
 
         mock_hook.return_value.create_spreadsheet.assert_called_once_with(spreadsheet=spreadsheet)
 
-        calls = [
-            mock.call(context, "spreadsheet_id", SPREADSHEET_ID),
-            mock.call(context, "spreadsheet_url", SPREADSHEET_URL),
-        ]
-        mock_xcom.has_calls(calls)
+        assert op_execute_result['spreadsheetId'] == '1234567890'
+        assert op_execute_result['spreadsheetUrl'] == 'https://example/sheets'
```
Comment (uranusjr) [inline tests/providers/google/suite/operators/test_sheets.py:47]: Does `op_execute_result` contain other values? We should try to verify the _entire_ dict if possible.

Comment (harishkrao) [reply tests/providers/google/suite/operators/test_sheets.py:47]: Thank you, I will write the entire dict's results and write more tests wherever appropriate.

Comment (harishkrao) [reply tests/providers/google/suite/operators/test_sheets.py:47]: I just printed the dict entirely and saw that these are the only two values the `GoogleSheetsCreateSpreadsheetOperator`'s execute call returns. 
The dict looks like this: `{'spreadsheetId': '1234567890', 'spreadsheetUrl': 'https://example/sheets'}`.

**Review by potiuk — APPROVED** [review]

**Pull request merged**